filebeat/module/postgresql/log/ingest/pipeline.json

```diff
@@ -0,0 +1,38 @@
+{
+  "description": "Pipeline for parsing PostgreSQL logs.",
+  "processors": [
+    {
+      "grok": {
+        "field": "message",
+        "ignore_missing": true,
+        "patterns": [
+          "%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timeformat} \[%{NUMBER:postgresql.log.thread_id}\] %{USERNAME:postgresql.log.username}@%{HOSTNAME:postgresql.log.host} %{WORD:postgresql.log.level}:  ?%{GREEDYDATA:postgresql.log.message}",
+          "%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timeformat} \[%{NUMBER:postgresql.log.thread_id}\] \[%{USERNAME:postgresql.username}\]@\[%{HOSTNAME:postgresql.log.host}\] %{WORD:postgresql.log.level}:  ?%{GREEDYDATA:postgresql.log.message}",
+          "%{LOCALDATETIME:postgresql.log.timestamp} %{WORD:postgresql.log.timeformat} \[%{NUMBER:postgresql.log.thread_id}\] %{WORD:postgresql.log.level}:  ?%{GREEDYDATA:postgresql.log.message}"
```

Comment:
I wonder if you need to use `\\` here. grok is basically build on regular expressions. Plus this file being json might might affect the `\` on read.